### PR TITLE
Add PhasedFSimCharacterization container that describes cirq.PhasedFSimGate unitary angles

### PR DESCRIPTION
**Pull request by mrwojtek**
This change adds PhasedFSimCharacterization that is a container for all the angles of cirq.PhasedFSimGate.

It is used for calibration purposes and stores only a numerical values for the angles, Since some calibrations estimate only a subset of the parameters, all of them are optional.

### CODE REVIEW

File: cirq/google/calibration/phased_fsim.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+if TYPE_CHECKING:
+    # Workaround for mypy custom dataclasses
+    from dataclasses import dataclass as json_serializable_dataclass
+else:
+    from cirq.protocols import json_serializable_dataclass
+
+
+@json_serializable_dataclass(frozen=True)
+class PhasedFSimParameters:
```
Comment (dstrain115) [inline cirq/google/calibration/phased_fsim.py:13]: Missing docstring

**Review by dstrain115 — COMMENTED** [review]
Is this specific to focused calibration?  I think it probably does belong in cirq.google, at least until it provides value outside of this.

**Comment by mrwojtek** [comment]
Hello cirq-maintainers, this is now ready for review.

File: cirq/google/calibration/phased_fsim.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
+from typing import Dict, Optional, TYPE_CHECKING
+
+import dataclasses
+
+if TYPE_CHECKING:
+    # Workaround for mypy custom dataclasses
```
Comment (balopat) [inline cirq/google/calibration/phased_fsim.py:6]: Can you just add the mypy tracking issue (https://github.com/python/mypy/issues/5406) to this comment?

**Review by balopat — APPROVED** [review]
Looks pretty good to me. @dstrain115?

File: cirq/google/calibration/phased_fsim.py (hunk 2 above)
Comment (mrwojtek) [reply cirq/google/calibration/phased_fsim.py]: Done.

File: cirq/google/calibration/phased_fsim.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
+from typing import Dict, Optional, TYPE_CHECKING
```
Comment (balopat) [inline cirq/google/calibration/phased_fsim.py:1]: This file is missing Copyright info. Can you add it?

File: cirq/google/calibration/phased_fsim_test.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+from cirq.google.calibration.phased_fsim import PhasedFSimCharacterization
```
Comment (balopat) [inline cirq/google/calibration/phased_fsim_test.py:1]: This too is missing copyright.

Comment (balopat) [reply cirq/google/calibration/phased_fsim_test.py:1]: I guess, we should have a script to check the boilerplate.

File: cirq/google/calibration/phased_fsim.py (hunk 3 above)
Comment (mrwojtek) [reply cirq/google/calibration/phased_fsim.py:15]: Done

File: cirq/google/calibration/phased_fsim_test.py (hunk 4 above)
Comment (mrwojtek) [reply cirq/google/calibration/phased_fsim_test.py:15]: Done

**Review by dstrain115 — APPROVED** [review]
Yeah, LGTM from me.

**Comment by mpharrigan** [comment]
I know this is already merged, but LGTM

**Pull request merged**